File: api/pokemon.js
export default async function handler(req, res) {
    const { q } = req.query;
    const apiKey = 'd18f5a1e-b4db-49ad-841b-6809c5f0515c'; 

    if (!q) return res.status(400).json({ error: 'Termo vazio' });

    // Tentativa 1: Busca exata pelo ID do set (ex: sv8a)
    // Tentativa 2: Busca por cartas que COMECEM com esse ID (ex: sv8a-001)
    // Tentativa 3: Busca pelo nome do set (ex: Terastal Festival)
    const query = `(set.id:"${q}" OR id:${q}* OR set.name:"*${q}*")`;
    const url = `https://api.pokemontcg.io/v2/cards?q=${encodeURIComponent(query)}&pageSize=100&select=id,name,number,images,set`;

    try {
        const response = await fetch(url, {
            headers: { 'X-Api-Key': apiKey }
        });
        const data = await response.json();

        res.setHeader('Access-Control-Allow-Origin', '*');
        res.setHeader('Content-Type', 'application/json');
        
        return res.status(200).json(data);
    } catch (error) {
        return res.status(500).json({ error: 'Erro na API' });
    }
}

File: api/pokemon.py
from http.server import BaseHTTPRequestHandler
import requests
import json
from urllib.parse import urlparse, parse_qs

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        query_params = parse_qs(urlparse(self.path).query)
        set_id = query_params.get("set", [""])[0].strip()
        
        # Se não houver set_id, evitamos o erro 500 retornando erro 400
        if not set_id:
            self.send_response(400)
            self.end_headers()
            self.wfile.write(b'{"error": "Parametro set e obrigatorio"}')
            return

        # URL oficial da Documentação para listagem por Set
        api_url = f"https://api.pokemonpricetracker.com/v1/cards"
        params = {
            "set_id": set_id,
            "limit": 250
        }

        try:
            # Fazendo a requisição com Timeout para não travar a Vercel
            response = requests.get(api_url, params=params, timeout=15)
            
            # Verifica se a API retornou erro (ex: 404 ou 403)
            response.raise_for_status()
            data = response.json()

            self.send_response(200)
            self.send_header('Content-type', 'application/json')
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            self.wfile.write(json.dumps(data).encode('utf-8'))
            
        except Exception as e:
            # Se der erro, ele envia a mensagem real para o seu console do navegador
            self.send_response(500)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            error_msg = {"error": "Falha na API Price Tracker", "details": str(e)}
            self.wfile.write(json.dumps(error_msg).encode('utf-8'))